File: airesources/Scala/src/main/scala/hlt/Player.scala
package hlt

class Player(val id: Short, ships: Map[Long, Ship]) {

  def getShip(entityId: Long): Option[Ship] = ships.get(entityId)

  override def toString: String = s"Player($getId, ${getShips.size})"

  def getShips: Map[Long, Ship] = ships

  def getId: Short = id
}

File: airesources/Rust/src/hlt/player.rs
use hlt::entity::Ship;
use hlt::parse::Decodable;

#[derive(PartialEq, Debug)]
pub struct Player {
    pub id: i32,
    pub ships: Box<[Ship]>,
}

impl Player {
    pub fn all_ships(&self) -> &[Ship] {
        &self.ships
    }
}

impl Decodable for Player {
    fn parse<'a, I>(tokens: &mut I) -> Self
    where
        I: Iterator<Item = &'a str>,
    {

        let id = i32::parse(tokens);
        let ships = Box::parse(tokens);

        Self { id, ships }
    }
}
